Sources/AppCoinsSDK/Domain/Repositories/Attribution/AttributionRepository.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  AttributionRepository.swift
+//
+//
+//  Created by Graciano Caldeira on 12/07/2024.
+//
+
+import Foundation
+
+internal class AttributionRepository: AttributionRepositoryProtocol {
+    
+    private let AttributionService: AttributionService = AttributionClient()
+    private var GuestUIDCache: Cache<String, Attribution> = Cache(cacheName: "Attribution")
+    
+    internal func getAttribution(completion: @escaping (Result<Attribution, Error>) -> Void) {
+        if let attribution = self.GuestUIDCache.getValue(forKey: "attribution") {
+            completion(.success(attribution))
+        } else {
+            self.AttributionService.getAttribution(bundleID: Bundle.main.bundleIdentifier ?? "") { result in
+                switch result {
+                case .success(let attributionRaw):
```

Comment:
We need to store both the GuestUID and the OEMID on UserDefaults, and we need to send them the next time we make the attribution request